src/Rings/MPolyMap/MPolyAnyMap.jl

```diff
@@ -250,7 +250,14 @@ end
 # Julia functions in both maps
 function compose(F::MPolyAnyMap{D, C, <: Function}, G::MPolyAnyMap{C, E, <: Function}) where {D, C, E}
   @req codomain(F) === domain(G) "Incompatible (co)domain in composition"
-  return hom(domain(F), codomain(G), x -> coefficient_map(G)(coefficient_map(F)(x)), G.(_images(F)), check=false)
+  b = coefficient_map(F)(one(coefficient_ring(domain(F))))
+  if parent(b) === domain(G)
+    return hom(domain(F), codomain(G), x -> G(coefficient_map(F)(x)), G.(_images(F)), check=false)
+  elseif parent(b) === coefficient_ring(domain(G))
+    return hom(domain(F), codomain(G), x -> coefficient_map(G)(coefficient_map(F)(x)), G.(_images(F)), check=false)
+  else
+    error("coefficient map is not admissible")
+  end
```

Comment:
@thofma : I think there was a bug here. Please check that this is now correct.

Reply:
Which input made it fail?

Reply:
When the coefficient_map of `F` takes values in the domain of `G`. This scenario was legitimate as far as I recall. The tests added here fail in case you remove this fix.

Reply:
Looks good. Can you add a test for the else branch to `test/Rings/MPolyAnyMap/MPolyRing`? It seems to not be covered.

Reply:
Done.